src/modules/flow/http-client/http-client.c

```diff
@@ -0,0 +1,600 @@
+        (struct sol_flow_node_type_http_client_boolean_options *)options;
+
+    mdata->url = strdup(opts->url);
+    SOL_NULL_CHECK(mdata->url, -EINVAL);
+
+    sol_ptr_vector_init(&mdata->pending_conns);
+
+    return 0;
+}
+
+static int
+common_url_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,
+    const struct sol_flow_packet *packet)
+{
+    int r;
+    const char *url;
+    struct http_data *mdata = data;
+
+    r = sol_flow_packet_get_string(packet, &url);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mdata->url = strdup(url);
+    SOL_NULL_CHECK(mdata->url, -ENOMEM);
+
+    return 0;
+}
+
+static int
+common_get_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,
+    const struct sol_flow_packet *packet)
+{
+    int r;
+    struct sol_http_param params;
+    struct http_data *mdata = data;
+    struct sol_http_client_connection *connection;
+    const struct http_client_node_type *type;
+
+    SOL_NULL_CHECK(mdata->url, -EINVAL);
+
+    sol_http_param_init(&params);
+    if (!sol_http_param_add(&params,
+        SOL_HTTP_REQUEST_PARAM_HEADER("Accept", "application/json"))) {
+        SOL_WRN("Failed to set query params");
+        sol_http_param_free(&params);
+        return -ENOMEM;
+    }
+
+    type = (const struct http_client_node_type *)sol_flow_node_get_type(node);
+    connection = sol_http_client_request(SOL_HTTP_METHOD_GET, mdata->url,
+        &params, type->request_finished_cb, node);
+
+    sol_http_param_free(&params);
+
+    SOL_NULL_CHECK(connection, -ENOTCONN);
+
+    r = sol_ptr_vector_append(&mdata->pending_conns, connection);
+    if (r < 0) {
+        SOL_WRN("Failed to keep pending connection.");
+        sol_http_client_connection_cancel(connection);
+        return -ENOMEM;
+    }
+
+    return 0;
+}
+
+static int
+check_response(struct http_data *mdata, struct sol_flow_node *node,
+    const struct sol_http_client_connection *connection,
+    struct sol_http_response *response)
+{
+    if (sol_ptr_vector_remove(&mdata->pending_conns, connection) < 0)
+        SOL_WRN("Failed to find pending connection %p", connection);
+
+    if (!response) {
+        sol_flow_send_error_packet(node, EINVAL,
+            "Error while reaching %s", mdata->url);
+        return -EINVAL;
+    }
+    SOL_HTTP_RESPONSE_CHECK_API(response, -EINVAL);
+
+    if (!response->content.used) {
+        sol_flow_send_error_packet(node, EINVAL,
+            "Empty response from %s", mdata->url);
+        return -EINVAL;
+    }
+
+    if (response->response_code != SOL_HTTP_STATUS_OK) {
+        sol_flow_send_error_packet(node, EINVAL,
+            "%s returned an unknown response code: %d",
+            mdata->url, response->response_code);
+        return -EINVAL;
+    }
+
+    return 0;
+}
+
+static int
+check_key(const char *url, struct sol_json_token *token)
+{
+    char *ptr;
+
+    if (strncmp(url, "http://", sizeof("http://") - 1) == 0)
```

Comment:
so far, our server does not support https, but yes, i'll check for it before commit. Thanks